tests/ProjectMembershipRolesTest.php

```diff
@@ -0,0 +1,309 @@
+                [
+                    'email' => 'spam@keboola.com',
+                    'role' => self::ROLE_GUEST,
+                ]
+            );
+            $this->fail('Action should not be allowed to guest users');
+        } catch (ClientException $e) {
+            $this->restrictedActionTest($e);
+        }
+
+        $this->assertCount(0, $this->normalUserWithMfaClient->listProjectInvitations($this->project['id']));
+    }
+
+    public function testGuestAdministratorCannotCancelAdministratorInvitation()
+    {
+        $this->assertCount(0, $this->normalUserWithMfaClient->listProjectInvitations($this->project['id']));
+
+        $invitation = $this->normalUserWithMfaClient->inviteUserToProject(
+            $this->project['id'],
+            [
+                'email' => 'spam@keboola.com',
+                'role' => self::ROLE_GUEST,
+            ]
+        );
+
+        try {
+            $this->guestRoleMemberClient->cancelProjectInvitation($this->project['id'], $invitation['id']);
+            $this->fail('Action should not be allowed to guest users');
+        } catch (ClientException $e) {
+            $this->restrictedActionTest($e);
+        }
+
+        $this->assertCount(1, $this->normalUserWithMfaClient->listProjectInvitations($this->project['id']));
+    }
+
+    public function testGuestAdministratorCanListProjectUsers()
+    {
+        $members = $this->guestRoleMemberClient->listProjectUsers($this->project['id']);
+        $this->assertCount(2, $members);
+    }
+
+    public function testGuestAdministratorCannotAddAdministrator()
+    {
+        try {
+            $this->guestRoleMemberClient->addUserToProject(
+                $this->project['id'],
+                [
+                    'email' => 'spam@keboola.com',
+                    'role' => self::ROLE_GUEST,
+                ]
+            );
+            $this->fail('Action should not be allowed to guest users');
+        } catch (ClientException $e) {
+            $this->restrictedActionTest($e);
+        }
+
+        $membership = $this->findProjectUser($this->project['id'], 'spam@keboola.com');
+        $this->assertNull($membership);
+    }
+
+    public function testGuestAdministratorCannotRemoveAdministrator()
+    {
+        try {
+            $this->guestRoleMemberClient->removeUserFromProject(
+                $this->project['id'],
+                $this->normalUserWithMfa['email']
+            );
+            $this->fail('Action should not be allowed to guest users');
+        } catch (ClientException $e) {
+            $this->restrictedActionTest($e);
+        }
+
+
+        $membership = $this->findProjectUser($this->project['id'], $this->normalUserWithMfa['email']);
+        $this->assertNotNull($membership);
+    }
+
+    public function testGuestAdministratorCanLeaveProject()
+    {
+        $this->guestRoleMemberClient->removeUserFromProject(
+            $this->project['id'],
+            $this->normalUser['id']
```

Comment:
možná bych změnil `$this->normalUser` na `$this->guestUser` trochu mě to tady zmátlo.

`$this->guestUser = $this->normalUser;` ?

Reply:
👍